ambry-commons/src/test/java/com.github.ambry.commons/ByteBufferReadableStreamChannelTest.java

```diff
@@ -28,33 +31,59 @@
 import java.util.concurrent.TimeoutException;
 import java.util.concurrent.atomic.AtomicBoolean;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
 import static org.junit.Assert.*;
 
 
 /**
  * Tests functionality of {@link ByteBufferReadableStreamChannel}.
  */
+@RunWith(Parameterized.class)
 public class ByteBufferReadableStreamChannelTest {
+  public static final int BUFFER_SIZE = 256;
+
+  private ByteBuffer contentWrapper;
+  private ByteBufferReadableStreamChannel byteBufferReadableStreamChannel;
+
+  public ByteBufferReadableStreamChannelTest(List<ByteBuffer> bufferList) {
+    if (bufferList.size() == 1) {
+      byteBufferReadableStreamChannel = new ByteBufferReadableStreamChannel(bufferList.get(0));
+    } else {
+      byteBufferReadableStreamChannel = new ByteBufferReadableStreamChannel(bufferList);
+    }
+    contentWrapper = ByteBuffer.allocate((int) byteBufferReadableStreamChannel.getSize());
+    for (ByteBuffer buffer: bufferList) {
+      contentWrapper.put(buffer.duplicate());
+    }
+    contentWrapper.rewind();
+  }
+
+  @Parameterized.Parameters
+  public static List<Object[]> data() {
+    // @formatter:off
+    return Arrays.asList(new Object[][] {
+      {Arrays.asList(ByteBuffer.wrap(TestUtils.getRandomBytes(BUFFER_SIZE)))},
```

Comment:
You could keep this simple and just pass `{true, false}` as you seem to have to do a check in the constructor anyway.

Also, do you think there a value in randomizing (with a reasonable max) the number of buffers in the list, rather than hardcoding it to 2?